app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,84 @@
+    val subsetToSubtract = setOf(bookMarkChild1)
+    val expectedRemainingSubset = listOf(bookMarkChild2)
+    val bookmarkNode = newBookmarkNode("Parent 1", 0, allChildren)
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a sub set of children THEN the children subset is removed and rest remains`() {
+        val bookmarkNodeSubsetRemoved = bookmarkNode.minus(subsetToSubtract)
+        assertEquals(expectedRemainingSubset, bookmarkNodeSubsetRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a set of all children THEN all children are removed`() {
+        val bookmarkNodeAllChildrenRemoved = bookmarkNode.minus(setofAllChildren)
+        assertEquals(emptyList<BookmarkNode>(), bookmarkNodeAllChildrenRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a set of non-children THEN no children are removed`() {
+        val bookMarkChild3 = newBookmarkNode("Child 3", 3, null)
+        val setofNonChildren = setOf(bookMarkChild3)
+        val bookmarkNodeNonChildrenRemoved = bookmarkNode.minus(setofNonChildren)
+        assertEquals(allChildren, bookmarkNodeNonChildrenRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing an empty set THEN no children are removed`() {
+        val bookmarkNodeEmptySetRemoved = bookmarkNode.minus(emptySet())
+        assertEquals(allChildren, bookmarkNodeEmptySetRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with an empty list as children WHEN minusing a set of non-children from an empty parent THEN an empty list remains`() {
+        val parentWithEmptyList = newBookmarkNode("Parent 1", 0, emptyList<BookmarkNode>())
+        val parentWithEmptyListNonChildRemoved = parentWithEmptyList.minus(setofAllChildren)
+        assertEquals(emptyList<BookmarkNode>(), parentWithEmptyListNonChildRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with null as children WHEN minusing a set of non-children from a parent with null children THEN null remains`() {
+        val parentWithNullList = newBookmarkNode("Parent 1", 0, null)
+        val parentWithNullListNonChildRemoved = parentWithNullList.minus(setofAllChildren)
+        assertEquals(null, parentWithNullListNonChildRemoved.children)
+    }
+
+    @Test
+    fun `GIVEN a bookmark node with children WHEN minusing a sub-set of children THEN the rest of the parents object should remain the same`() {
+        val resultBookmarkNode = bookmarkNode.minus(subsetToSubtract)
+        val restOfResult = resultBookmarkNode.copy(children = expectedRemainingSubset)
+        val restofOriginal = bookmarkNode.copy(children = expectedRemainingSubset)
+        assertEquals(restOfResult, restofOriginal)
+    }
+
+    private fun newBookmarkNode(name: String, id: Int, myChildren: List<BookmarkNode>?) = BookmarkNode(
+            BookmarkNodeType.ITEM,
+            "11",
```

Comment:
nit: when reading the code without an IDE (such as for these reviews), it's helpful to include the argument names: `value = ...`

Also, when you have multiple args in a row with the same type (e.g. strings), it's pretty easy to mess them up, which is another reason to add explicit names.